stac_mjx/stac.py

```diff
@@ -339,15 +351,20 @@ def mjx_setup(kp_data, mj_model):
         )
 
         # q_phase - root
-        if self._mj_model.jnt_type[0] == mujoco.mjtJoint.mjJNT_FREE:
+        if self._root_kp_idx == -1:
+            print(
+                "Missing or invalid ROOT_OPTIMIZATION_KEYPOINT, skipping root_optimization()"
+            )
+        elif self._mj_model.jnt_type[0] == mujoco.mjtJoint.mjJNT_FREE:
             vmap_root_opt = jax.vmap(
                 compute_stac.root_optimization,
-                in_axes=(0, 0, 0, None, None, None, None),
+                in_axes=(0, 0, 0, None, None, None, None, None),
```

Comment:
did you figure out why it wasn't working before?

Reply:
Before? I added new variable to root_optimization, root_kp_idx, so had to make a change here. Is that what you mean?

Reply:
I remember you tried adding that root_kp_idx feature before but it wouldn't compile properly? So we had to revert it.

Reply:
Yeah, I think I misunderstanding what was happening, and that it was another change that was causing that issue.